NuKeeper.Inspection/RepositoryInspection/DirectoryBuildTargetsReader.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using NuKeeper.Abstractions;
+using NuKeeper.Abstractions.Logging;
+
+namespace NuKeeper.Inspection.RepositoryInspection
+{
+    public class DirectoryBuildTargetsReader : IPackageReferenceFinder
+
+    {
+        private readonly INuKeeperLogger _logger;
+
+        public DirectoryBuildTargetsReader(INuKeeperLogger logger)
+        {
+            _logger = logger;
+        }
+
+        public IReadOnlyCollection<PackageInProject> ReadFile(string baseDirectory, string relativePath)
+        {
+            var packagePath = new PackagePath(baseDirectory, relativePath, PackageReferenceType.DirectoryBuildTargets);
+            try
+            {
+                using (var fileContents = File.OpenRead(packagePath.FullName))
+                {
+                    return Read(fileContents, packagePath);
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new NuKeeperException($"Unable to parse file {packagePath.FullName}", ex);
+            }
+        }
+
+        public IReadOnlyCollection<string> GetFilePatterns()
+        {
+            return new[] {"Directory.Build.props", "Directory.Build.targets"};
+        }
+
+        public IReadOnlyCollection<PackageInProject> Read(Stream fileContents, PackagePath path)
+        {
+            var xml = XDocument.Load(fileContents);
+
+            var packagesNode = xml.Element("Project")?.Element("ItemGroup");
+            if (packagesNode == null)
+            {
+                return Array.Empty<PackageInProject>();
+            }
+
+            var packageNodeList = packagesNode.Elements()
+                .Where(x => x.Name == "PackageReference");
```

Comment:
Will the target, build file only work with those exact definitions? So only "PackageReference" and not "packageReference" for example?

Reply:
It's XML, after all. Yes it's case sensitive. Because of MSBuild quirks (and dotnetcore history), one could also use nodes instead of attributes for `Version`, but no sample I've seen has done that, so I'm only sticking to the current preferred syntax.

Reply:
To expand a bit: this file is a normal MSBuild script, in the end, so... it's pretty much free form. We can only rely on the convention to handle the most common cases.

Reply:
Great